lib/mongoid/extensions/set.rb

```diff
@@ -37,9 +37,13 @@ def demongoize(object)
         #
         # @param [ Set ] object The object to mongoize.
         #
-        # @return [ Array ] The object mongoized.
+        # @return [ Array | nil ] The object mongoized or nil.
         def mongoize(object)
-          object.to_a
+          return if object.nil?
+          case object
+          when ::Set then ::Array.mongoize(object.to_a)
```

Comment:
Shouldn't this line also have a `uniq` at the end in case the mongoizer maps multiple inputs to the same output?

Reply:
Yes

Reply:
done